pyteal/ast/router.py

```diff
@@ -913,22 +927,27 @@ def compile_program(
             * clear_state_program: compiled clear-state program string
             * contract: a Python SDK Contract object to allow clients to make off-chain calls
         """
-        ap, csp, contract = self._build_program(version=version, optimize=optimize)
-        ap_compiled = compileTeal(
-            ap,
-            Mode.Application,
-            version=version,
-            assembleConstants=assemble_constants,
-            optimize=optimize,
-        )
-        csp_compiled = compileTeal(
-            csp,
-            Mode.Application,
-            version=version,
-            assembleConstants=assemble_constants,
-            optimize=optimize,
-        )
-        return ap_compiled, csp_compiled, contract
+        starting_slot_id = ScratchSlot.nextSlotId
+        try:
```

Comment:
nit: I am not too happy about using `try...finally` here. Is it possible to somehow find a hidden boolean indicator to say the current router is "contaminated" or not, then raise error here?

Reply:
filing a pr here #649